aicsimageio/readers/mrc_reader.py

```diff
@@ -0,0 +1,149 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+from typing import Any, Tuple
+
+import dask.array as da
+import mrcfile
+import numpy as np
+import xarray as xr
+from fsspec.implementations.local import LocalFileSystem
+from fsspec.spec import AbstractFileSystem
+from numpy.lib.recfunctions import structured_to_unstructured
+
+from .. import constants, exceptions, types
+from ..dimensions import DimensionNames
+from ..metadata.utils import generate_ome_image_id
+from ..utils import io_utils
+from .reader import Reader
+
+
+class MrcReader(Reader):
+    """
+    Read MRC files using mrcfile.
+
+    Parameters
+    ----------
+    image: types.PathLike
+        Path to image file to construct Reader for.
+    """
+
+    @staticmethod
+    def _is_supported_image(fs: AbstractFileSystem, path: str, **kwargs: Any) -> bool:
+        try:
+            with mrcfile.open(path, permissive=True, header_only=True):
+                return True
+        except (ValueError, TypeError):
+            return False
+
+    def __init__(self, image: types.PathLike):
+        self._physical_pixel_sizes = None
+        # Expand details of provided image
+        self._fs, self._path = io_utils.pathlike_to_fs(image, enforce_exists=True)
+        # Catch non-local file system
+        if not isinstance(self._fs, LocalFileSystem):
+            raise NotImplementedError(
+                f"mrc reader not yet implemented for non-local file system. "
+                f"Received URI: {self._path}, which points to {type(self._fs)}."
+            )
+
+        if not self._is_supported_image(self._fs, self._path):
+            raise exceptions.UnsupportedFileFormatError(
+                self.__class__.__name__, self._path
+            )
```

Comment:
This is super out of the scope of this PR, but I definitely think we should have a shared function for this since it seems to be so common :smile:

Reply:
Yup, I just copy pasted from eslewhere :)